src/cpp/include/openvino/genai/text2image/pipeline.hpp

```diff
@@ -120,6 +122,14 @@ class OPENVINO_GENAI_EXPORTS Text2ImagePipeline {
         const UNet2DConditionModel& unet,
         const AutoencoderKL& vae_decoder);
 
+    // creates SDXL pipeline from building blocks
+    static Text2ImagePipeline stable_diffusion_xl(
+        const std::shared_ptr<Scheduler>& scheduler_type,
+        const CLIPTextModel& clip_text_model,
+        const CLIPTextModelWithProjection& clip_text_model_with_projection,
```

Comment:
text encoder 2 can be None https://huggingface.co/Kwai-Kolors/Kolors/blob/main/model_index.json

Let's consider such case un future PRs as well